ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java

```diff
@@ -100,11 +101,13 @@ void registerDisconnect(final PeerConnection connection) {
 
   private void abortPendingRequestsAssignedToDisconnectedPeers() {
     synchronized (this) {
-      pendingRequests.stream()
-          .filter(
-              pendingPeerRequest ->
-                  pendingPeerRequest.getAssignedPeer().map(EthPeer::isDisconnected).orElse(false))
-          .forEach(PendingPeerRequest::abort);
+      final Iterator<PendingPeerRequest> iterator = pendingRequests.iterator();
```

Comment:
Obviously it seems that a copy like this can also result in a ConcurrentModificationException if a deletion is done during the copy

I found this solution and I did some tests. it's working fine 
> private final Collection<PendingPeerRequest> pendingRequests = new CopyOnWriteArrayList<>();